File: cli/src/main/kotlin/com/maestrorecorder/cli/parser/CoordinateConverter.kt
package com.maestrorecorder.cli.parser

class CoordinateConverter(
    private val inputMaxX: Int,
    private val inputMaxY: Int,
    private val screenWidth: Int,
    private val screenHeight: Int
) {
    fun toPixelX(rawX: Int): Int =
        (rawX.toLong() * screenWidth / inputMaxX).toInt()

    fun toPixelY(rawY: Int): Int =
        (rawY.toLong() * screenHeight / inputMaxY).toInt()
}

File: packages/server/src/recording/coordinate-converter.ts
export class CoordinateConverter {
  constructor(
    private inputMaxX: number,
    private inputMaxY: number,
    private screenWidth: number,
    private screenHeight: number,
  ) {}

  toPixelX(rawX: number): number {
    return Math.floor((rawX * this.screenWidth) / this.inputMaxX);
  }

  toPixelY(rawY: number): number {
    return Math.floor((rawY * this.screenHeight) / this.inputMaxY);
  }

  /** Convert tap distance threshold from pixels to raw sensor units */
  pixelsToRawDistance(pixels: number): number {
    return Math.floor((pixels * this.inputMaxX) / this.screenWidth);
  }
}
